src/sentry/models/groupassignee.py

```diff
@@ -42,6 +43,7 @@ def assign(self, group, assigned_to, acting_user=None):
             )
         else:
             affected = True
+            issue_assigned.send(organization=group.project.organization, project=group.project, group=group, sender=acting_user)
```

Comment:
User-triggered, no new db calls.

This could potentially (via API) be triggered more often, but it's not part of the event processing pipeline, so I'm inclined to say that performance isn't critical here.